lib/active_record/virtual_attributes/virtual_reflections.rb

```diff
@@ -9,26 +9,24 @@ module ClassMethods
         # Definition
         #
 
-        def virtual_has_one(name, options = {})
-          uses = options.delete(:uses)
+        def virtual_has_one(name, uses: nil, **options)
           reflection = ActiveRecord::Associations::Builder::HasOne.build(self, name, nil, options)
-          add_virtual_reflection(reflection, name, uses, options)
+          add_virtual_reflection(reflection, name, uses)
         end
 
-        def virtual_has_many(name, options = {})
+        def virtual_has_many(name, uses: nil, source: nil, through: nil, **options)
           define_method(:"#{name.to_s.singularize}_ids") do
             records = send(name)
             records.respond_to?(:ids) ? records.ids : records.collect(&:id)
           end
-          uses = options.delete(:uses)
           reflection = ActiveRecord::Associations::Builder::HasMany.build(self, name, nil, options)
-          add_virtual_reflection(reflection, name, uses, options)
+          add_virtual_reflection(reflection, name, uses)
+          define_delegate(name, source || name, :to => through, :allow_nil => true, :default => []) if through
```

Comment:
Should this method return the virtual reflection like it did before because now it's changed to return the delegate (I presume add_virtual_reflection returns the reflection).

Can we define the delegate before the add_virtual_reflection? If so, we can just swap these 2 lines.